src/libs/protocols/protos/oci.proto

```diff
@@ -161,11 +161,20 @@ message Hooks {
 	// Prestart is a list of hooks to be run before the container process is executed.
 	repeated Hook Prestart = 1  [(gogoproto.nullable) = false];
 
+	// CreateRuntime is a list of hooks to be run during the creation of runtime(sandbox).
+	repeated Hook CreateRuntime = 2  [(gogoproto.nullable) = false];
+
+	// CreateContainer is a list of hooks to be run after VM is started, and before container is created.
+	repeated Hook CreateContainer = 3  [(gogoproto.nullable) = false];
+
+	// StartContainer is a list of hooks to be run after container is created, but before it is started.
+	repeated Hook StartContainer = 4  [(gogoproto.nullable) = false]; 
+
 	// Poststart is a list of hooks to be run after the container process is started.
-	repeated Hook Poststart = 2  [(gogoproto.nullable) = false];
+	repeated Hook Poststart = 5  [(gogoproto.nullable) = false];
 
 	// Poststop is a list of hooks to be run after the container process exits.
-	repeated Hook Poststop = 3  [(gogoproto.nullable) = false];
+	repeated Hook Poststop = 6  [(gogoproto.nullable) = false];
```

Comment:
When editing agent protocols, please do not insert things in the middle of a struct. It will break agent protocol compatibility (which is already in a bad state.)

Reply:
OK! I will fix it ASAP.